File: filemanager/src/main/java/com/merseyside/filemanager/mime_types/ImageTypes.java
package com.merseyside.filemanager.mime_types;


import android.mtp.MtpConstants;

public enum ImageTypes {

    JPEG(new MimeType(getFormat(), "jpeg", MtpConstants.FORMAT_EXIF_JPEG)),
    JPG(new MimeType(getFormat(), "jpeg", MtpConstants.FORMAT_EXIF_JPEG)),
    GIF(new MimeType(getFormat(), "gif", MtpConstants.FORMAT_GIF)),
    PNG(new MimeType(getFormat(), "png", MtpConstants.FORMAT_PNG)),
    BMP(new MimeType(getFormat(), "x-ms-bmp", MtpConstants.FORMAT_BMP));

    MimeType type;

    ImageTypes(MimeType type) {
        this.type = type;
    }

    public MimeType getValue() {
        return type;
    }

    public static String getFormat() {
        return "image";
    }
}

File: filemanager/src/main/java/com/merseyside/filemanager/mime_types/ImageTypes.kt
package com.merseyside.filemanager.mime_types

import android.mtp.MtpConstants

enum class ImageTypes(var value: MimeType) {
    JPEG(
        MimeType(format, "jpeg", MtpConstants.FORMAT_EXIF_JPEG)
    ),
    JPG(
        MimeType(
            format,
            "jpeg",
            MtpConstants.FORMAT_EXIF_JPEG
        )
    ),
    GIF(
        MimeType(format, "gif", MtpConstants.FORMAT_GIF)
    ),
    PNG(
        MimeType(format, "png", MtpConstants.FORMAT_PNG)
    ), BMP(
        MimeType(
            format,
            "x-ms-bmp",
            MtpConstants.FORMAT_BMP
        )
    );

    companion object {
        val format: String
            get() = "image"
    }
}